Library/Homebrew/command.rb

```diff
@@ -94,8 +106,8 @@ def self.argv_options_without_value_passed(argv_options_only)
         .select { |opt| valid_options_with_values.map { |x| x[:option] }.include?(opt.split("=", 2)[0]) }
         .select do |opt|
           (opt.include?("=") && opt.split("=", 2)[1].empty?) ||
-            (!opt.include?("=") && ARGV[ARGV.index(opt)+1] &&
-              ARGV[ARGV.index(opt)+1].start_with?("-"))
+            (!opt.include?("=") && (ARGV[ARGV.index(opt)+1].nil? ||
```

Comment:
Command classes should probably duplicate `ARGV` into their own instance field called `argv` (or something along those lines), since the ultimate goal is to move away from directly accessing and operating on `ARGV`.